File: backend/modules/dtos/response/quiz_result.go
package response

import (
	"time"

	"github.com/SamuelJacobsenB/projeto-the_sophium/back/types"
)

type QuizResultDTO struct {
	ID           string             `json:"id"`
	EnrollmentID string             `json:"enrollment_id"`
	QuizID       string             `json:"quiz_id"`
	Attempts     int                `json:"attempts"`
	Corrections  []types.Correction `json:"corrections"`
	CreatedAt    time.Time          `json:"created_at"`
	UpdatedAt    time.Time          `json:"updated_at"`
}

File: frontend/src/types/entities/quiz_result.ts
export interface Correction {
  question_index: number;
  selected: number;
  correct_index: number;
}

export interface QuizResult {
  id: string;
  enrollment_id: string;
  quiz_id: string;
  attempts: number;
  corrections: Correction[];
  created_at: Date;
  updated_at: Date;
}
